util/tracer_nvbit/tracer_tool/traces-processing/post-traces-processing.cpp

```diff
@@ -152,9 +165,31 @@ void group_per_block(const char *filepath) {
 	//ss.ignore(); //remove the space
 	//rest_of_line.clear();
       // getline(ss, rest_of_line); //get rest of the string!
-	string rest_of_line(ss.str().substr(ss.tellg()+1));
+	    string rest_of_line(ss.str().substr(ss.tellg()+1));
+
+      // Ni: ignore the shmem LDGSTS instruction
+      stringstream opcode_ss;
+      string opcode, temp;
+      unsigned dest_num;
+      opcode_ss << rest_of_line;
+      for (int i = 0; i < 2; i++) {
+        opcode_ss >> temp;
+      }
+      opcode_ss >> dest_num;
+      for (unsigned i = 0; i < dest_num; i++) {
+        opcode_ss >> temp;
+      }
+      opcode_ss >> opcode;
 
-      insts[tb_id].warp_insts_array[warpid_tb].push_back(rest_of_line);
+      if (opcode.find("LDGSTS") != string::npos) {
+        if (!LDGSTS_flag[tb_id][warpid_tb]) {
+          insts[tb_id].warp_insts_array[warpid_tb].push_back(rest_of_line);
+        }
+        LDGSTS_flag[tb_id][warpid_tb] = !LDGSTS_flag[tb_id][warpid_tb];
+      }

```

Comment:
Don't quite understand what's going on in the outer "if" block. Why flip the boolean flag?

Reply:
One actual LDGSTS instruction includes 2 LDGSTS instructions in the trace, because it has two memory references. This is trying to remove the one with the shared memory address.

Reply:
Please put that down explicitly in comments